src/trace/context.ts

```diff
@@ -55,6 +56,48 @@ export interface StepFunctionContext {
   "step_function.state_retry_count": number;
 }
 
+export function readTraceFromStepFunctionsContext(stepFunctionContext: StepFunctionContext): TraceContext | undefined {
+  return {
+    traceID: deterministicMd5HashToBigIntString(stepFunctionContext["step_function.execution_id"]),
+    parentID: deterministicMd5HashToBigIntString(
+      stepFunctionContext["step_function.execution_id"] +
+        "#" +
+        stepFunctionContext["step_function.state_name"] +
+        "#" +
+        stepFunctionContext["step_function.state_entered_time"],
+    ),
+    sampleMode: SampleMode.AUTO_KEEP.valueOf(),
+    source: Source.Event,
+  };
+}
+
+export function hexToBinary(hex: string) {
+  // convert hex to binary and padding with 0 in the front to fill 128 bits
+  return parseInt(hex, 16).toString(2).padStart(4, "0");
+}
+
+export function deterministicMd5HashInBinary(s: string): string {
+  // Md5 here is not used as an encryption method but to generate a deterministic hash as the backend does
```

Comment:
Pedantic edit, since hashing != encryption
```suggestion
  // Md5 here is used here because we don't need a cryptographically secure hashing method but to generate the same trace/span ids as the backend does
```